### PR TITLE
Add Authentication with NextAuth

### LINKED ISSUE #1: Set up auth using Pesu-Auth
Use NextAuth and create a custom provider to use https://github.com/HackerSpace-PESU/pesu-auth to authenticate new users

### PR DESCRIPTION
**Pull request by sixtusagbo**
I added a Sign In page at /signin
I also setup a custom provider for authentication with PESU Auth

resolves #1 
closes #1

### CODE REVIEW

**Comment by sixtusagbo** [comment]
Please don't merge yet.

Hi maintainers, I need PESU Auth credentials to test it out. One of you can send it [to my email](mailto:miracleagbosixtus@gmail.com) or [on Discord](https://discordapp.com/users/947899643608588328).

File: src/pages/signin.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import { signIn } from "next-auth/react";
+import { FormEvent, FunctionComponent, useState } from "react";
+import { Icons } from "~/components/Icons";
+
+const SignInPage: FunctionComponent = () => {
+  const [username, setUsername] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+  const [error, setError] = useState<null | string>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+
```
Comment (sixtusagbo) [inline src/pages/signin.tsx:10]: Hello

**Comment by rithulkamesh** [comment]
hey @sixtusagbo, unfortunately we cannot provide you with that since that's our official university credentials, however we can test it for you.

**Comment by sixtusagbo** [comment]
Nevermind, I'm mocking it already. Will push my changes soon. You were taking too long so I decided to mock it. When I push, you can then test with your credentials.

**Comment by sixtusagbo** [comment]
Your API does not return the user's id so I'm generating it with crypto.

**Comment by sixtusagbo** [comment]
You can test it now

**Comment by rithulkamesh** [comment]
the SRN serves as our ID, it's our global identification across the university.

**Comment by rithulkamesh** [comment]
@sixtusagbo can you update the ID and resolve the merge conflict from our page so that I can merge this? I've checked with my peers and your auth system works.

**Comment by silicoflare** [comment]
@sixtusagbo please fix the merge conflicts so that we can continue working on the login page and other features. Thanks!

**Comment by sixtusagbo** [comment]
Okay

**Comment by sixtusagbo** [comment]
I've fixed the merge conflict, just a minor issue.

**Comment by rithulkamesh** [comment]
Looks great, Thank you so much for your contribution

**Pull request merged**